contributed_definitions/nyaml/NXpeak.yaml

```diff
@@ -1,59 +1,140 @@
+  Base class for describing spectral peaks, their functional form, and support values
+  (i.e., the discretization (points) at which the function has been evaluated).
 symbols:
   doc: |
     The symbols used in the schema to specify e.g. dimensions of arrays.
   n_support: |
-    Number of support points
+    Number of support points.
 type: group
 NXpeak(NXobject):
   label:
     doc: |
-      Human-readable identifier to specify which concept/entity
+      Human-readable label which specifies which concept/entity
       the peak represents/identifies.
-  peak_model:
+  lineshape:
     doc: |
       Is the peak described analytically via a functional form
       or is it empirically defined via measured/reported
-      intensity/counts as a function of an independent variable.
+      intensity/counts as a function of an independent variable?
       
-      If the functional form is not empirical or gaussian, users
-      should enter other for the peak_model and add relevant details
+      If the functional form is not empirical or Gaussians, users
+      should enter other for the description and add relevant details
       in the NXcollection.
-    enumeration: [empirical, gaussian, lorentzian, other]
+    enumeration: [empirical, gaussian, lorentzian, voigt, other]
+  lineshape_other:
+    doc: |
+      Human-readable descritpion of the lineshape used for the peak if it does
+      not fit into the enumeration given in `lineshape`.
   position(NX_NUMBER):
     unit: NX_ANY
     doc: |
       In the case of an empirical description of the peak and its shoulders,
       this array holds the position values for the independent variable.
-    dimensions:
-      rank: 1
-      dim: [[1, n_support]]
+    dim: (n_support,)
   intensity(NX_NUMBER):
     unit: NX_ANY
     doc: |
       In the case of an empirical description of the peak and its shoulders,
       this array holds the intensity/count values at each position.
-    dimensions:
-      rank: 1
-      dim: [[1, n_support]]
-  (NXcollection):
+    dim: (n_support,)
+  width(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Width of a peak at a defined fraction of the peak height.
+      
+      Usually, this will be the Full Width at Half Maximum of the peak (FWHM).
+      For asymmetric peaks, convenient measures of peak width are the half-widths of
+      each side of the peak at half maximum intensity.
+    \@type:
+      doc: |
+        Specification of the definition of `width`.
+      enumeration:
+        fwhm:
+          doc: |
+            Full Width at Half Maximum of the peak.
+        hwhm:
+          doc: |
+            Half Width at Half Maximum of the peak.
+        half_widths_asymmetric:
+          doc: |
+            For asymmetrical peaks, the half-widths of each side of the peak at
+            half maximum intensity is given.
+        other:
+          doc: |
+            For other peak width definitions.
```

Comment:
Isn't that redundant with the peak_parameters?

Reply:
Probably yes. The main reason why I wanted to keep this is that in NXxps, I am referencing and ISO defintion for peak width. But, this is not strictly necessary, so I will remove it.